nsv13/code/modules/overmap/starmap.dm

```diff
@@ -74,10 +74,17 @@
 	var/datum/star_system/current_system = info["current_system"]
 	SSstar_system.update_pos(linked)
 	if(linked.ftl_drive)
-		data["ftl_progress"] = linked.ftl_drive.progress
-		if(linked.ftl_drive.ftl_state == FTL_STATE_READY)
-			data["ftl_progress"] = linked.ftl_drive.spoolup_time
-		data["ftl_goal"] = linked.ftl_drive.spoolup_time //TODO
+		if(istype(linked.ftl_drive))
```

Comment:
This is actually unecessary, as long as both objects have the same vars, you don't need to worry about typecasting at all. Look at my new shield component in boarding for an example of this beautiful hot garbage.

Reply:
@Kmc2000 I'm aware, I've done that for most ftl drive interactions but I needed an exception for this because they both use a different spool system. I've changed it up a bit since then so I might be able to remove this line, we'll see 😆